### ISSUE TITLE
Restore ORCID implementation

### ISSUE THREAD

**Issue by kaplun** [inspirehep/inspire-next#2798]
304cfcaead150ca5ab5e470dda5e51c3449550b8 removed the then current implementation of ORCID support. The first step to integrate ORCID into INSPIRE should be to restore that work and build/improve on it.

**Comment by jacquerie** [comment]
More realistically, this module has to be reimplemented from scratch, perhaps peeking at the previous implementation.

**Comment by kaplun** [comment]
I am not convinced about the *from scratch* part. It was left in a functional state but disabled by a feature flag because we were not yet ready to transition it from legacy (we still miss the opt-in/out functionality and the possibility to pick up a profile).
Also, IIRC, it was based on `invenio-orcid` which is also based on `python-orcid`, which was also based on the 2.0 RC version of the API not on 1.3 one.

**Comment by jacquerie** [comment]
> It was left in a functional state

No, the schema has changed a lot since, and yet the tests were still passing because they were mocking too aggressively too many things.

**Comment by kaplun** [comment]
Yes fine, so we need to adapt:
https://github.com/inspirehep/inspire-next/commit/304cfcaead150ca5ab5e470dda5e51c3449550b8#diff-976c02effc41b0da5f3f9ce9afd561c6 (i.e. schema.py). Doesn't mean we need to throw away everything, and redo from scratch.
